mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAOptimizer.scala

```diff
@@ -542,10 +539,11 @@ private[clustering] object OnlineLDAOptimizer {
       expElogbeta: BDM[Double],
       alpha: breeze.linalg.Vector[Double],
       gammaShape: Double,
-      k: Int): (BDV[Double], BDM[Double]) = {
-    val (ids: List[Int], cts: Array[Double]) = termCounts match {
-      case v: DenseVector => ((0 until v.size).toList, v.values)
-      case v: SparseVector => (v.indices.toList, v.values)
+      k: Int,
+      ids: List[Int]): (BDV[Double], BDM[Double]) = {
+    val cts: Array[Double] = termCounts match {
```

Comment:
Yes it looks redundant, but `values` is not a member of parent trait `Vector`, it is defined at individual implementation level (i.e.  `DenseVector` and `SparseVector`).